src/lp.rs

```diff
@@ -0,0 +1,428 @@
+                    .uri
+                    .as_ref()
+                    .map(|uri| format!("{}?context={}", uri, &lp_id));
+                embed = embed
+                    .title("Listening Party in full swing! Join in!")
+                    .field(
+                        "Now playing",
+                        format!(
+                            "Track {} - {} - {} / {}",
+                            track.number,
+                            maybe_uri(&track.name, track_uri_ctx.as_ref()),
+                            display_duration(&position),
+                            display_duration(&track.duration)
+                        ),
+                        true,
+                    );
+            }
+        }
+
+        Box::new(embed)
+    }
+}
+
+// Format Duration as hh:mm:ss
+fn display_duration(duration: &chrono::Duration) -> String {
+    let allsecs = duration.num_seconds();
+    let seconds = allsecs % 60;
+    let minutes = allsecs / 60 % 60;
+    let hours = allsecs / 3600;
+    if hours > 0 {
+        format!("{}:{:0>2}:{:0>2}", hours, minutes, seconds)
+    } else {
+        format!("{:0>2}:{:0>2}", minutes, seconds)
+    }
+}
+
+// Regex to identity spotify album URIs and extract album id
+const SPOTIFY_ALBUM_RE: &str =
+    "\\bhttps://open.spotify.com/album/([a-zA-Z0-9]+)(?:\\?[a-zA-Z?=&]*)\\b";
+
+const SPOTIFY_PLAYLIST_RE: &str =
+    "\\bhttps://open.spotify.com/playlist/([a-zA-Z0-9]+)(?:\\?[a-zA-Z?=&]*)\\b";
+
+#[derive(Command, Debug)]
+#[cmd(name = "lp", desc = "Check if listening party is going")]
+pub struct CurrentLP {}
+
+#[async_trait]
+impl BotCommand for CurrentLP {
+    type Data = Handler;
+    async fn run(
+        self,
+        data: &Handler,
+        _ctx: &Context,
+        interaction: &CommandInteraction,
+    ) -> anyhow::Result<CommandResponse> {
+        let channel = interaction.channel_id;
+        let lpmod = data.module::<LP>().unwrap();
+        let lps = lpmod.last_pinged.read().await;
+        let lp = lps.get(&channel);
+        let response = match lp {
+            None => CommandResponse::Public(
+                "There is no listening party at the moment.".to_string(),
+            ),
+            Some(lpinfo) => CommandResponse::Embed(lpinfo.build_embed()),
+        };
+        Ok(response)
+    }
+}
+
+pub type PingedMap = Arc<RwLock<HashMap<ChannelId, LPInfo>>>;
+
+pub struct LP {
+    last_pinged: PingedMap,
+}
+
+impl Clone for LP {
+    fn clone(&self) -> Self {
+        LP {
+            last_pinged: self.last_pinged.clone(),
+        }
+    }
+}
+
+#[async_trait]
+impl polls::ModPollReadyHandler for LP {
+    async fn ready(&self, channelid: &ChannelId) {
+        self.start_lp(channelid).await;
+    }
+}
+
+// Roles used for pinging listening parties
+const LP_ROLES: &'static [u64] = &[
+    1198354637137391709, // `@Listening Party`` in test guild
+                         // TODO: Make this configurable? Fetch via name?
+];
```

Comment:
The Lp module in serenity-command-handler defines a role_id guild field, could use that for consistency (it can be defined here too without risk of conflicts)

Reply:
I was thinking about this - the roles should ideally work both in test instances and on the deployed instance, perhaps they should be fetched by name when we connect to the guild, if that's possible?

Reply:
The way I've come up with now is to keep taking the role ID from the message (essentially free since they are provided by serenity anyway) and dynamically resolve them to their name via `to_role_cached`. This should be "cheap enough" since the request is cached and it only happens on role mentions.

Reply:
Hmm, maybe it should get the role from this: https://github.com/etwyniel/discord_framework/blob/f26138393edb495b7d951129f13cd04350a86c5a/serenity-command-handler/src/modules/lp.rs#L349-L376

However, this only seems to be a single role for listening parties that the `LP` module uses to ping.